src/lib.rs

```diff
@@ -262,19 +270,25 @@ impl Mnemonic {
 		Err(Error::AmbiguousWordList(langs))
 	}
 
-	/// Parse a mnemonic and detect the language from the enabled languages.
-	pub fn parse<'a, S: Into<Cow<'a, str>>>(s: S) -> Result<Mnemonic, Error> {
+	/// Parse a mnemonic in the given language.
+	pub fn parse_in<'a, S: Into<Cow<'a, str>>>(language: Language, s: S) -> Result<Mnemonic, Error> {
 		let mut cow = s.into();
 		Mnemonic::normalize_utf8_cow(&mut cow);
-		let language = Mnemonic::language_of(cow.as_ref())?;
 		Mnemonic::validate_in(language, cow.as_ref())?;
 		Ok(Mnemonic(cow.into_owned()))
 	}
 
-	/// Parse a mnemonic in the given language.
-	pub fn parse_in<'a, S: Into<Cow<'a, str>>>(language: Language, s: S) -> Result<Mnemonic, Error> {
+	/// Parse a mnemonic and detect the language from the enabled languages.
+	pub fn parse<'a, S: Into<Cow<'a, str>>>(s: S) -> Result<Mnemonic, Error> {
 		let mut cow = s.into();
 		Mnemonic::normalize_utf8_cow(&mut cow);
+
+		let language = if Language::all().len() == 1 {
```

Comment:
Is this still useful with the improved performance of `language_of`?

Reply:
Yeah well it's more efficient. Perhaps I'm overdoing performance a bit here :p